Add submission ui

iglesias294 opened the pull request:
Includes submission UI, backend S3 upload signing on index route.

iglesias294 committed:
Create submit-submission page

iglesias294 committed:
Create submit-submission page

iglesias294 committed:
Create submit-submission page

iglesias294 committed:
Create submit-submission page

iglesias294 committed:
Merge branch 'add_submission_ui' of https://github.com/hatchways/team-hummingbird into add_submission_ui

iglesias294 committed:
Add s3 uploader to UI & s3 signing service to backend

iglesias294 committed:
Add s3 uploader to UI & s3 signing service to backend

iglesias294 committed:
merged latest dev

iglesias294 committed:
Merge branch 'add_submission_ui' of https://github.com/hatchways/team-hummingbird into add_submission_ui

iglesias294 committed:
added upload previews, error handling

Renamed SubmitSubmission, added error handling, snackbar alerts for success and errors, backend call

iglesias294 committed:
Merge branch 'dev' into add_submission_ui

bonnieli commented on client/src/pages/SubmitSubmission.js:
```diff
@@ -0,0 +1,171 @@
+
+    //get userID from future state manager
+    const user_id = props.user_id
+    const contest_id = props.contest_id
```
keep consistent with snake_case and camelCase

camelCase is better for JS :)

bonnieli commented on client/src/pages/SubmitSubmission.js:
```diff
@@ -0,0 +1,171 @@
+                            <br />
+                            <Typography
+                                display="block"
+                                style={{ color: 'gray', fontWeight: 'lighter' }}
```
same styling as below, maybe put this in a class

iglesias294 committed:
remove axios, add removal button

uninstalled axios
replaced API call with fetch async await
replace underlined consts with camelCase
added unique number to name of every file uploaded
fixed multi-file uploads
submission upload preview now shows multiple previews
Instruction text add instructions for adding additional files when previews are loaded.
added button to previews for removing uploads

perryraskin commented:
This looks great to me
